dvc/utils/fs.py

```diff
@@ -95,15 +96,16 @@ def move(src, dst, mode=None):
     dst = os.path.abspath(dst)
     tmp = f"{dst}.{uuid()}"
 
+    with reraise(OSError, FileOwnershipError(src)):
```

Comment:
We could move that below `mode` check.

Reply:
EDIT: @danielfleischer I am sorry for messing up, I just realized that your original approach is better:
So, the fact that we catch `OSError` does not mean that its a problem with permissions. To make this error accurate, we should actually check for error code.
So, I would recommend rolling back to the first version of change, 
catch the `OSError as e` and then check whether the error code is not `errno.EPERM` (and possibly `errno.EACCES`)
Just like we are doing [here](https://github.com/iterative/dvc/blob/411b38896014ae004d0782e25dbd49f5d33af2b7/dvc/utils/fs.py#L118).
And only then raise `FileOwnershipError`
[Error codes reference](https://docs.python.org/3.8/library/errno.html)

Reply:
File not owned by user will lead to `errno.EPERM`. The question is are there more cases that will lead to permission error which are not related to ownership and then decide what's more helpful for the user: permission problem or ownership problem.